File: lib/lib_timesys/date2j.f
       
C Procedure           DATE2J
C
      INTEGER FUNCTION     DATE2J( YEAR, MONTH, DAY )
C
C
C                            Purpose
C
C  This integer function (calendar DATE 2 Julian date) takes the input
C  Gregorian calendar date and returns as its functional value the
C  corresponding integer Julian date. Since the Julian date is an intege
C  this correspondence is exact for noon of the output calendar date.
C
C  The algorithm for this conversion is taken from the following article
C  Tantzen,R.T., "Communications of the ACM", Volume 6, Number 8, August
C  Algorithm 199, page 444.
C
C
C                            Input_Arguments
C
C  YEAR     is the year number.
C  MONTH    is the month number.
C  DAY      is the day number.
C
C
C                            Output_Arguments
C
C                            Declarations_of_External_Functions
C
C                            Declarations_of_Input_and_Output_Arguments
C
      INTEGER            YEAR
      INTEGER            MONTH
      INTEGER            DAY
C
C
C                            Declarations_of_Local_Variables
C
      INTEGER            Y
      INTEGER            M
      INTEGER            D
C
      INTEGER            C
      INTEGER            YA
C
C
C                     Method
C-&
C***********************************************************************
C
      Y = YEAR
      M = MONTH
      D = DAY
C
      IF( M .GT. 2 )THEN
         M = M - 3
      ELSE
         M = M + 9
         Y = Y - 1
      END IF
C
      C  = Y/100
      YA = Y - 100*C
      DATE2J = (146097*C)/4 + (1461*YA)/4 + (153*M+2)/5 + D + 1721119
C
      RETURN
      END   

File: tool/GraceRead_L1_2008_03_20/lib/date2j.c
#include "TimeLib.h"

int date2j( int year, int month, int day )
{
  static char SccsId[] = "@(#) date2j.c       1.1 01/08/01";

  /* Purpose:
  This integer function (calendar DATE 2 Julian date) takes the input
  Gregorian calendar date and returns as its functional value the
  corresponding integer Julian date. Since the Julian date is an integer
  this correspondence is exact for noon of the output calendar date.

  The algorithm for this conversion is taken from the following article
  Tantzen,R.T., "Communications of the ACM", Volume 6, Number 8, August
  Algorithm 199, page 444. */

  /* Input_Arguments:

  year     is the year number.
  month    is the month number.
  day      is the day number.  */

  /* Output_Arguments */

  /* Declarations_of_External_Functions */
  /* none */

  /* Declarations_of_Local_Variables */

  int Y;
  int M;
  int D;

  int C;
  int YA;
      
  Y = year;
  M = month;
  D = day;

  if( M > 2 )
    M = M - 3;
  else
  {
    M = M + 9;
    Y = Y - 1;
  }

  C  = Y/100;
  YA = Y - 100*C;
  return ((146097*C)/4 + (1461*YA)/4 + (153*M+2)/5 + D + 1721119);

}
